dandiapi/api/models/asset_paths.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from django.db import models
+from django.db.models import Q
+from django_extensions.db.models import TimeStampedModel
+
+from dandiapi.api.models.asset import Asset
+from dandiapi.api.models.version import Version
+
+
+class AssetPath(TimeStampedModel):
+    path = models.CharField(max_length=512)
+
+    # TODO: Determine what should be done on delete
```

Comment:
I would think `models.CASCADE` would make sense here, right? If an `Asset` is deleted, there's no reason to keep any of its `AssetPath`s around any longer.

Reply:
It's tricky, since if we did that it would allow for unilaterally deleting an asset, taking the asset paths with them, but without updating the asset paths tree. E.g., if we deleted an asset with that was `5kb` in size, all parent paths would have their `aggregate_size` field +5kb larger than it should be, as well as the `aggregate_files` 1 file larger than it should be.

I'm actually inclined to to `models.PROTECT` here, since it would enforce that you need to call `delete_asset_paths` before deleting the asset.

Reply:
Ah, right. Yeah I agree with using `models.PROTECT` to enforce that `delete_asset_paths` is called before deleting an asset.

Reply:
https://github.com/dandi/dandi-archive/pull/1312/commits/76eb5ee1915d6fa1d68776415ec17cfbe3682e8a